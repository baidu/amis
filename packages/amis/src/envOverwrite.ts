/**
 * @file 用于在移动端或不同语言环境下使用不同配置
 */

import {SchemaNode, Schema} from './types';
import {RendererProps, RendererConfig, addSchemaFilter} from './factory';
import {findObjectsWithKey} from './utils/helper';
const isMobile = (window as any).matchMedia?.('(max-width: 768px)').matches
  ? true
  : false;

// 这里不能用 addSchemaFilter 是因为还需要更深层的替换，比如 select 里的 options
export const envOverwrite = (schema: any, locale?: string) => {
  if (schema.mobile && isMobile) {
    Object.assign(schema, schema.mobile);
    delete schema.mobile;
  }

  if (locale) {
    let schemaNodes = findObjectsWithKey(schema, locale);
    for (let schemaNode of schemaNodes) {
      Object.assign(schemaNode, schemaNode[locale]);
      delete schemaNode[locale];
    }
  }

  if (isMobile) {
    let schemaNodes = findObjectsWithKey(schema, 'mobile');
    for (let schemaNode of schemaNodes) {
      Object.assign(schemaNode, schemaNode['mobile']);
      delete schemaNode['mobile'];
    }
  }
};
