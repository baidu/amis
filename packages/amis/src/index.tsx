/** @license amis v@version
 *
 * Copyright Baidu
 *
 * This source code is licensed under the Apache license found in the
 * LICENSE file in the root directory of this source tree.
 */
export * from 'amis-core';
export * from 'amis-ui';

import './preset';

// 注册渲染器
import './renderers/Action';
import './renderers/Alert';
import './renderers/App';
import './renderers/Avatar';
import './renderers/Remark';
import './renderers/ButtonGroup';
import './renderers/Form/ButtonToolbar';
import './renderers/Breadcrumb';
import './renderers/DropDownButton';
import './renderers/Calendar';
import './renderers/Collapse';
import './renderers/CollapseGroup';
import './renderers/Color';
import './renderers/CRUD';
import './renderers/CRUD2';
import './renderers/Pagination';
import './renderers/Cards';
import './renderers/Card';
import './renderers/Card2';
import './renderers/Custom';
import './renderers/Date';
import './renderers/Dialog';
import './renderers/Divider';
import './renderers/Each';
import './renderers/Flex';
import './renderers/Form/Control';
import './renderers/Form/Hidden';
import './renderers/Form/InputText';
import './renderers/Form/InputTag';
import './renderers/Form/InputNumber';
import './renderers/Form/Textarea';
import './renderers/Form/Checkboxes';
import './renderers/Form/Checkbox';
import './renderers/Form/InputCity';
import './renderers/Form/ChartRadios';
import './renderers/Form/InputRating';
import './renderers/Form/Switch';
import './renderers/Form/Radios';
import './renderers/Form/Radio';
import './renderers/Form/JSONSchema';
import './renderers/Form/JSONSchemaEditor';
import './renderers/Form/ListSelect';
import './renderers/Form/LocationPicker';
import './renderers/Form/Select';
import './renderers/Form/Static';
import './renderers/Form/InputDate';
import './renderers/Form/InputDateRange';
import './renderers/Form/InputFormula';
import './renderers/Form/InputRepeat';
import './renderers/Form/InputTree';
import './renderers/Form/TreeSelect';
import './renderers/Form/InputImage';
import './renderers/Form/InputFile';
import './renderers/Form/UUID';
import './renderers/Form/MatrixCheckboxes';
import './renderers/Form/InputMonthRange';
import './renderers/Form/InputQuarterRange';
import './renderers/Form/InputYearRange';
import './renderers/Form/InputRange';
import './renderers/Form/InputArray';
import './renderers/Form/Combo';
import './renderers/Form/ConditionBuilder';
import './renderers/Form/InputSubForm';
import './renderers/Form/InputExcel';
import './renderers/Form/InputRichText';
import './renderers/Form/Editor';
import './renderers/Form/DiffEditor';
import './renderers/Form/InputColor';
import './renderers/Form/ChainedSelect';
import './renderers/Form/NestedSelect';
import './renderers/Form/Transfer';
import './renderers/Form/TransferPicker';
import './renderers/Form/InputTable';
import './renderers/Form/Picker';
import './renderers/Form/IconPicker';
import './renderers/Form/IconSelect';
import './renderers/Form/Formula';
import './renderers/Form/FieldSet';
import './renderers/Form/TabsTransfer';
import './renderers/Form/TabsTransferPicker';
import './renderers/Form/Group';
import './renderers/Form/InputGroup';
import './renderers/Form/UserSelect';
import './renderers/Grid';
import './renderers/Grid2D';
import './renderers/HBox';
import './renderers/VBox';
import './renderers/Image';
import './renderers/Images';
import './renderers/List';
import './renderers/Log';
import './renderers/Operation';
import './renderers/Page';
import './renderers/PaginationWrapper';
import './renderers/Panel';
import './renderers/Plain';
import './renderers/Property';
import './renderers/Portlet';
import './renderers/Spinner';
import './renderers/Table/index';
import './renderers/Tabs';
import './renderers/Tpl';
import './renderers/Mapping';
import './renderers/Progress';
import './renderers/Status';
import './renderers/Json';
import './renderers/Link';
import './renderers/Wizard';
import './renderers/Chart';
import './renderers/Container';
import './renderers/SwitchContainer';
import './renderers/SearchBox';
import './renderers/Service';
import './renderers/SparkLine';
import './renderers/Video';
import './renderers/Audio';
import './renderers/Nav';
import './renderers/Number';
import './renderers/Tasks';
import './renderers/Drawer';
import './renderers/Wrapper';
import './renderers/IFrame';
import './renderers/BarCode';
import './renderers/QRCode';
import './renderers/Icon';
import './renderers/Carousel';
import './renderers/AnchorNav';
import './renderers/Steps';
import './renderers/Timeline';
import './renderers/Markdown';
import './renderers/TableView';
import './renderers/Code';
import './renderers/WebComponent';
import './renderers/GridNav';
import './renderers/TooltipWrapper';
import './renderers/Tag';
import './renderers/Table2/index';
import './renderers/Words';
import './renderers/Password';
import './renderers/DateRange';
import './renderers/MultilineText';
import './renderers/OfficeViewer';
import './renderers/AMIS';

import './compat';
import './schemaExtend';
import type {
  BaseSchema,
  FormSchema,
  SchemaApi,
  SchemaCollection,
  SchemaExpression,
  SchemaObject
} from './Schema';
import type {TableViewSchema, TrObject} from './renderers/TableView';
import type {ActionSchema, ButtonSchema} from './renderers/Action';
import type {CRUDCommonSchema} from './renderers/CRUD';
import type {CRUD2Schema} from './renderers/CRUD2';
import type {TabsSchema} from './renderers/Tabs';
import {availableLanguages as EditorAvailableLanguages} from './renderers/Form/Editor';
import type {Action} from './types';
export * from './renderers/Form/IconPickerIcons';
export * from './renderers/Form/IconSelectStore';

export {
  BaseSchema,
  SchemaCollection,
  FormSchema,
  SchemaApi,
  SchemaObject,
  TableViewSchema,
  TrObject,
  ActionSchema,
  CRUDCommonSchema,
  ButtonSchema,
  CRUD2Schema,
  TabsSchema,
  SchemaExpression,
  Action,
  EditorAvailableLanguages
};
