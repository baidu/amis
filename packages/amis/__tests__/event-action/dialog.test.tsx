import {fireEvent, render, waitFor} from '@testing-library/react';
import '../../src';
import {render as amisRender} from '../../src';
import {makeEnv, wait} from '../helper';

test('1. EventAction:dialog args', async () => {
  const notify = jest.fn();
  const {getByText, container}: any = render(
    amisRender(
      {
        type: 'page',
        body: [
          {
            type: 'button',
            label: '打开弹窗',
            onEvent: {
              click: {
                actions: [
                  {
                    actionType: 'dialog',
                    args: {
                      dialog: {
                        type: 'dialog',
                        id: 'dialog_001',
                        title: '模态弹窗',
                        body: [
                          {
                            type: 'button',
                            label: '打开子弹窗',
                            onEvent: {
                              click: {
                                actions: [
                                  {
                                    actionType: 'dialog',
                                    args: {
                                      dialog: {
                                        type: 'dialog',
                                        title: '模态子弹窗',
                                        body: [
                                          {
                                            type: 'button',
                                            label: '关闭父弹窗',
                                            onEvent: {
                                              click: {
                                                actions: [
                                                  {
                                                    actionType: 'closeDialog',
                                                    componentId: 'dialog_001'
                                                  }
                                                ]
                                              }
                                            }
                                          }
                                        ]
                                      }
                                    }
                                  }
                                ]
                              }
                            }
                          },
                          {
                            type: 'button',
                            label: '关闭当前弹窗',
                            className: 'ml-2',
                            onEvent: {
                              click: {
                                actions: [
                                  {
                                    actionType: 'closeDialog'
                                  }
                                ]
                              }
                            }
                          },
                          {
                            type: 'button',
                            label: '触发确认',
                            className: 'ml-2',
                            onEvent: {
                              click: {
                                actions: [
                                  {
                                    actionType: 'confirm',
                                    componentId: 'dialog_001'
                                  }
                                ]
                              }
                            }
                          },
                          {
                            type: 'button',
                            label: '触发取消',
                            className: 'ml-2',
                            onEvent: {
                              click: {
                                actions: [
                                  {
                                    actionType: 'cancel',
                                    componentId: 'dialog_001'
                                  }
                                ]
                              }
                            }
                          }
                        ],
                        onEvent: {
                          confirm: {
                            actions: [
                              {
                                actionType: 'toast',
                                args: {
                                  msg: 'confirm'
                                }
                              }
                            ]
                          },
                          cancel: {
                            actions: [
                              {
                                actionType: 'toast',
                                args: {
                                  msg: 'cancel'
                                }
                              }
                            ]
                          }
                        }
                      }
                    }
                  }
                ]
              }
            }
          }
        ]
      },
      {},
      makeEnv({
        getModalContainer: () => container,
        notify
      })
    )
  );

  // events
  fireEvent.click(getByText('打开弹窗'));
  expect(container).toMatchSnapshot();

  await waitFor(() => {
    expect(getByText('确认')).toBeInTheDocument();
  });
  fireEvent.click(getByText('确认'));
  await wait(300);
  expect(notify).toHaveBeenCalled();
  await waitFor(() => {
    expect(container.querySelector('[role="dialog"]')).not.toBeInTheDocument();
  });
  expect(container).toMatchSnapshot();

  fireEvent.click(getByText('打开弹窗'));
  await waitFor(() => {
    expect(getByText('取消')).toBeInTheDocument();
  });
  fireEvent.click(getByText('取消'));
  await wait(300);
  expect(notify).toHaveBeenCalled();
  await waitFor(() => {
    expect(container.querySelector('[role="dialog"]')).not.toBeInTheDocument();
  });
  expect(container).toMatchSnapshot();

  // actions
  fireEvent.click(getByText('打开弹窗'));
  await waitFor(() => {
    expect(getByText('关闭当前弹窗')).toBeInTheDocument();
  });
  fireEvent.click(getByText('关闭当前弹窗'));
  await waitFor(() => {
    expect(container.querySelector('[role="dialog"]')).not.toBeInTheDocument();
  });
  expect(container).toMatchSnapshot();

  fireEvent.click(getByText('打开弹窗'));
  await waitFor(() => {
    expect(getByText('打开子弹窗')).toBeInTheDocument();
  });
  fireEvent.click(getByText('打开子弹窗'));
  await waitFor(() => {
    expect(getByText('关闭父弹窗')).toBeInTheDocument();
  });
  fireEvent.click(getByText('关闭父弹窗'));
  await waitFor(() => {
    expect(container.querySelector('[role="dialog"]')).not.toBeInTheDocument();
  });
  expect(container).toMatchSnapshot();

  fireEvent.click(getByText('打开弹窗'));
  await waitFor(() => {
    expect(getByText('触发确认')).toBeInTheDocument();
  });
  fireEvent.click(getByText('触发确认'));
  await wait(300);
  expect(notify).toHaveBeenCalled();
  await waitFor(() => {
    expect(container.querySelector('[role="dialog"]')).not.toBeInTheDocument();
  });
  expect(container).toMatchSnapshot();

  fireEvent.click(getByText('打开弹窗'));
  await waitFor(() => {
    expect(getByText('触发取消')).toBeInTheDocument();
  });
  fireEvent.click(getByText('触发取消'));
  await wait(300);
  expect(notify).toHaveBeenCalled();
  await waitFor(() => {
    expect(container.querySelector('[role="dialog"]')).not.toBeInTheDocument();
  });
  expect(container).toMatchSnapshot();
}, 7000);

test('2. EventAction:dialog', async () => {
  const notify = jest.fn();
  const {getByText, container}: any = render(
    amisRender(
      {
        type: 'page',
        body: [
          {
            type: 'button',
            label: '打开弹窗',
            onEvent: {
              click: {
                actions: [
                  {
                    actionType: 'dialog',
                    dialog: {
                      type: 'dialog',
                      id: 'dialog_001',
                      title: '模态弹窗',
                      body: [
                        {
                          type: 'button',
                          label: '打开子弹窗',
                          onEvent: {
                            click: {
                              actions: [
                                {
                                  actionType: 'dialog',
                                  dialog: {
                                    type: 'dialog',
                                    title: '模态子弹窗',
                                    body: [
                                      {
                                        type: 'button',
                                        label: '关闭父弹窗',
                                        onEvent: {
                                          click: {
                                            actions: [
                                              {
                                                actionType: 'closeDialog',
                                                componentId: 'dialog_001'
                                              }
                                            ]
                                          }
                                        }
                                      }
                                    ]
                                  }
                                }
                              ]
                            }
                          }
                        },
                        {
                          type: 'button',
                          label: '关闭当前弹窗',
                          className: 'ml-2',
                          onEvent: {
                            click: {
                              actions: [
                                {
                                  actionType: 'closeDialog'
                                }
                              ]
                            }
                          }
                        },
                        {
                          type: 'button',
                          label: '触发确认',
                          className: 'ml-2',
                          onEvent: {
                            click: {
                              actions: [
                                {
                                  actionType: 'confirm',
                                  componentId: 'dialog_001'
                                }
                              ]
                            }
                          }
                        },
                        {
                          type: 'button',
                          label: '触发取消',
                          className: 'ml-2',
                          onEvent: {
                            click: {
                              actions: [
                                {
                                  actionType: 'cancel',
                                  componentId: 'dialog_001'
                                }
                              ]
                            }
                          }
                        }
                      ],
                      onEvent: {
                        confirm: {
                          actions: [
                            {
                              actionType: 'toast',
                              args: {
                                msg: 'confirm'
                              }
                            }
                          ]
                        },
                        cancel: {
                          actions: [
                            {
                              actionType: 'toast',
                              args: {
                                msg: 'cancel'
                              }
                            }
                          ]
                        }
                      }
                    }
                  }
                ]
              }
            }
          }
        ]
      },
      {},
      makeEnv({
        getModalContainer: () => container,
        notify
      })
    )
  );

  // events
  fireEvent.click(getByText('打开弹窗'));
  expect(container).toMatchSnapshot();

  await waitFor(() => {
    expect(getByText('确认')).toBeInTheDocument();
  });
  fireEvent.click(getByText('确认'));
  await wait(300);
  expect(notify).toHaveBeenCalled();
  await waitFor(() => {
    expect(container.querySelector('[role="dialog"]')).not.toBeInTheDocument();
  });
  expect(container).toMatchSnapshot();

  fireEvent.click(getByText('打开弹窗'));
  await waitFor(() => {
    expect(getByText('取消')).toBeInTheDocument();
  });
  fireEvent.click(getByText('取消'));
  await wait(300);
  expect(notify).toHaveBeenCalled();
  await waitFor(() => {
    expect(container.querySelector('[role="dialog"]')).not.toBeInTheDocument();
  });
  expect(container).toMatchSnapshot();

  // actions
  fireEvent.click(getByText('打开弹窗'));
  await waitFor(() => {
    expect(getByText('关闭当前弹窗')).toBeInTheDocument();
  });
  fireEvent.click(getByText('关闭当前弹窗'));
  await waitFor(() => {
    expect(container.querySelector('[role="dialog"]')).not.toBeInTheDocument();
  });
  expect(container).toMatchSnapshot();

  fireEvent.click(getByText('打开弹窗'));
  await waitFor(() => {
    expect(getByText('打开子弹窗')).toBeInTheDocument();
  });
  fireEvent.click(getByText('打开子弹窗'));
  await waitFor(() => {
    expect(getByText('关闭父弹窗')).toBeInTheDocument();
  });
  fireEvent.click(getByText('关闭父弹窗'));
  await waitFor(() => {
    expect(container.querySelector('[role="dialog"]')).not.toBeInTheDocument();
  });
  expect(container).toMatchSnapshot();

  fireEvent.click(getByText('打开弹窗'));
  await waitFor(() => {
    expect(getByText('触发确认')).toBeInTheDocument();
  });
  fireEvent.click(getByText('触发确认'));
  await wait(300);
  expect(notify).toHaveBeenCalled();
  await waitFor(() => {
    expect(container.querySelector('[role="dialog"]')).not.toBeInTheDocument();
  });
  expect(container).toMatchSnapshot();

  fireEvent.click(getByText('打开弹窗'));
  await waitFor(() => {
    expect(getByText('触发取消')).toBeInTheDocument();
  });
  fireEvent.click(getByText('触发取消'));
  await wait(300);
  expect(notify).toHaveBeenCalled();
  await waitFor(() => {
    expect(container.querySelector('[role="dialog"]')).not.toBeInTheDocument();
  });
  expect(container).toMatchSnapshot();
}, 7000);

test('3. EventAction:dialog data', async () => {
  const {getByText, container}: any = render(
    amisRender(
      {
        type: 'page',
        data: {
          name: 'amis'
        },
        body: [
          {
            type: 'button',
            label: '打开弹窗',
            onEvent: {
              click: {
                actions: [
                  {
                    actionType: 'dialog',
                    args: {
                      dialog: {
                        type: 'dialog',
                        id: 'dialog_001',
                        title: '模态弹窗${event.data.name}',
                        body: [
                          {
                            type: 'tpl',
                            tpl: '你好，我是${name}、${name1}、${name2}'
                          }
                        ],
                        data: {
                          name1: '${name}',
                          name2: '${event.data.name}'
                        }
                      }
                    }
                  }
                ]
              }
            }
          }
        ]
      },
      {},
      makeEnv({
        getModalContainer: () => container
      })
    )
  );

  // events
  fireEvent.click(getByText('打开弹窗'));
  await waitFor(() => {
    expect(container.querySelector('[role="dialog"]')).toBeInTheDocument();
    expect(getByText('模态弹窗')).toBeInTheDocument();
    expect(getByText('你好，我是、amis、amis')).toBeInTheDocument(); // 因为事件动作给args提前做了映射
  });
}, 7000);

test('4. EventAction:dialog data2', async () => {
  const {getByText, container}: any = render(
    amisRender(
      {
        type: 'page',
        data: {
          name: 'amis'
        },
        body: [
          {
            type: 'button',
            label: '打开弹窗',
            onEvent: {
              click: {
                actions: [
                  {
                    actionType: 'dialog',
                    dialog: {
                      type: 'dialog',
                      id: 'dialog_001',
                      title: '模态弹窗${event.data.name}',
                      body: [
                        {
                          type: 'tpl',
                          tpl: '你好，我是${name}、${name1}、${name2}'
                        }
                      ],
                      data: {
                        name1: '${name}',
                        name2: '${event.data.name}'
                      }
                    }
                  }
                ]
              }
            }
          }
        ]
      },
      {},
      makeEnv({
        getModalContainer: () => container
      })
    )
  );

  // events
  fireEvent.click(getByText('打开弹窗'));
  await waitFor(() => {
    expect(container.querySelector('[role="dialog"]')).toBeInTheDocument();
    expect(getByText('模态弹窗')).toBeInTheDocument();
    expect(getByText('你好，我是、amis、amis')).toBeInTheDocument();
  });
}, 7000);

// // test('EventAction:alert', async () => {
// //   const alert = jest.fn();
// //   const {getByText, container}: any = render(
// //     amisRender(
// //       {
// //         type: 'page',
// //         data: {
// //           msg: '去吃饭了'
// //         },
// //         body: [
// //           {
// //             type: 'button',
// //             label: '提示对话框',
// //             level: 'primary',
// //             onEvent: {
// //               click: {
// //                 actions: [
// //                   {
// //                     actionType: 'alert',
// //                     args: {
// //                       msg: '<a href="http://www.baidu.com" target="_blank">${msg}~</a>'
// //                     }
// //                   }
// //                 ]
// //               }
// //             }
// //           }
// //         ]
// //       },
// //       {},
// //       makeEnv({
// //         getModalContainer: () => container,
// //         alert
// //       })
// //     )
// //   );

// //   fireEvent.click(getByText('提示对话框'));
// //   await waitFor(() => {
// //     expect(alert).toHaveBeenCalled();
// //   });
// //   expect(alert.mock.calls[0][0]).toEqual(
// //     '<a href="http://www.baidu.com" target="_blank">去吃饭了~</a>'
// //   );
// // });

// // test('EventAction:confirm', async () => {
// //   const confirm = jest.fn();
// //   const {getByText, container}: any = render(
// //     amisRender(
// //       {
// //         type: 'page',
// //         data: {
// //           title: '操作确认',
// //           msg: '要删除它吗？'
// //         },
// //         body: [
// //           {
// //             type: 'button',
// //             label: '确认对话框',
// //             level: 'primary',
// //             onEvent: {
// //               click: {
// //                 actions: [
// //                   {
// //                     actionType: 'confirmDialog',
// //                     args: {
// //                       title: '${title}',
// //                       msg: '<span style="color:red">${msg}</span>'
// //                     }
// //                   }
// //                 ]
// //               }
// //             }
// //           }
// //         ]
// //       },
// //       {},
// //       makeEnv({
// //         getModalContainer: () => container,
// //         confirm
// //       })
// //     )
// //   );

// //   fireEvent.click(getByText('确认对话框'));
// //   await waitFor(() => {
// //     expect(confirm).toHaveBeenCalled();
// //   });
// //   expect(confirm.mock.calls[0][0]).toEqual(
// //     '<span style="color:red">要删除它吗？</span>'
// //   );
// //   expect(confirm.mock.calls[0][1]).toEqual('操作确认');
// // });

test('5. EventAction:dialog formitem without form', async () => {
  const onAction = jest.fn();
  const {getByText, container}: any = render(
    amisRender(
      {
        type: 'page',
        body: [
          {
            type: 'button',
            label: 'Dialog',
            onEvent: {
              click: {
                actions: [
                  {
                    actionType: 'dialog',
                    dialog: {
                      title: '表单页面',
                      body: [
                        {
                          label: 'A',
                          type: 'input-text',
                          name: 'a'
                        },
                        {
                          label: 'B',
                          type: 'input-text',
                          name: 'b'
                        }
                      ],
                      onEvent: {
                        confirm: {
                          actions: [
                            {
                              actionType: 'custom',
                              script: onAction
                            }
                          ]
                        }
                      }
                    }
                  }
                ]
              }
            }
          }
        ]
      },
      {},
      makeEnv({
        getModalContainer: () => container
      })
    )
  );

  const button = getByText('Dialog');
  fireEvent.click(button);
  await wait(200);

  fireEvent.change(container.querySelector('[name="a"]')!, {
    target: {value: '1'}
  });

  await wait(200);
  fireEvent.change(container.querySelector('[name="b"]')!, {
    target: {value: '2'}
  });

  await wait(200);
  fireEvent.click(getByText('确认'));
  await wait(300);
  expect(onAction).toHaveBeenCalled();
  expect(onAction.mock.calls[0][2].data).toMatchObject({a: '1', b: '2'});
});
