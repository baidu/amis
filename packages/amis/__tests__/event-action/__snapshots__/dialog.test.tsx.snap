// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. EventAction:dialog args 1`] = `
<div>
  <div
    class="cxd-Page"
  >
    <div
      class="cxd-Page-content"
    >
      <div
        class="cxd-Page-main"
      >
        <div
          class="cxd-Page-body"
          role="page-body"
        >
          <button
            class="cxd-Button cxd-Button--default cxd-Button--size-default"
            type="button"
          >
            <span>
              打开弹窗
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="amis-dialog-widget cxd-Modal cxd-Modal--1th"
    role="dialog"
  >
    <div
      class="cxd-Modal-overlay in"
    />
    <div
      class="cxd-Modal-content in"
    >
      <div
        class="cxd-Modal-header"
      >
        <a
          class="cxd-Modal-close"
          data-position="left"
          data-tooltip="关闭"
        >
          <icon-mock
            classname="icon icon-close"
            icon="close"
          />
        </a>
        <div
          class="cxd-Modal-title"
        >
          模态弹窗
        </div>
      </div>
      <div
        class="cxd-Modal-body"
        role="dialog-body"
      >
        <button
          class="cxd-Button cxd-Button--default cxd-Button--size-default"
          type="button"
        >
          <span>
            打开子弹窗
          </span>
        </button>
        <button
          class="cxd-Button cxd-Button--default cxd-Button--size-default ml-2"
          type="button"
        >
          <span>
            关闭当前弹窗
          </span>
        </button>
        <button
          class="cxd-Button cxd-Button--default cxd-Button--size-default ml-2"
          type="button"
        >
          <span>
            触发确认
          </span>
        </button>
        <button
          class="cxd-Button cxd-Button--default cxd-Button--size-default ml-2"
          type="button"
        >
          <span>
            触发取消
          </span>
        </button>
      </div>
      <div
        class="cxd-Modal-footer"
      >
        <button
          class="cxd-Button cxd-Button--default cxd-Button--size-default"
          type="button"
        >
          <span>
            取消
          </span>
        </button>
        <button
          class="cxd-Button cxd-Button--primary cxd-Button--size-default"
          type="button"
        >
          <span>
            确认
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`1. EventAction:dialog args 2`] = `
<div>
  <div
    class="cxd-Page"
  >
    <div
      class="cxd-Page-content"
    >
      <div
        class="cxd-Page-main"
      >
        <div
          class="cxd-Page-body"
          role="page-body"
        >
          <button
            class="cxd-Button cxd-Button--default cxd-Button--size-default"
            type="button"
          >
            <span>
              打开弹窗
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`1. EventAction:dialog args 3`] = `
<div>
  <div
    class="cxd-Page"
  >
    <div
      class="cxd-Page-content"
    >
      <div
        class="cxd-Page-main"
      >
        <div
          class="cxd-Page-body"
          role="page-body"
        >
          <button
            class="cxd-Button cxd-Button--default cxd-Button--size-default"
            type="button"
          >
            <span>
              打开弹窗
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`1. EventAction:dialog args 4`] = `
<div>
  <div
    class="cxd-Page"
  >
    <div
      class="cxd-Page-content"
    >
      <div
        class="cxd-Page-main"
      >
        <div
          class="cxd-Page-body"
          role="page-body"
        >
          <button
            class="cxd-Button cxd-Button--default cxd-Button--size-default"
            type="button"
          >
            <span>
              打开弹窗
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`1. EventAction:dialog args 5`] = `
<div>
  <div
    class="cxd-Page"
  >
    <div
      class="cxd-Page-content"
    >
      <div
        class="cxd-Page-main"
      >
        <div
          class="cxd-Page-body"
          role="page-body"
        >
          <button
            class="cxd-Button cxd-Button--default cxd-Button--size-default"
            type="button"
          >
            <span>
              打开弹窗
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`1. EventAction:dialog args 6`] = `
<div>
  <div
    class="cxd-Page"
  >
    <div
      class="cxd-Page-content"
    >
      <div
        class="cxd-Page-main"
      >
        <div
          class="cxd-Page-body"
          role="page-body"
        >
          <button
            class="cxd-Button cxd-Button--default cxd-Button--size-default"
            type="button"
          >
            <span>
              打开弹窗
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`1. EventAction:dialog args 7`] = `
<div>
  <div
    class="cxd-Page"
  >
    <div
      class="cxd-Page-content"
    >
      <div
        class="cxd-Page-main"
      >
        <div
          class="cxd-Page-body"
          role="page-body"
        >
          <button
            class="cxd-Button cxd-Button--default cxd-Button--size-default"
            type="button"
          >
            <span>
              打开弹窗
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`2. EventAction:dialog 1`] = `
<div>
  <div
    class="cxd-Page"
  >
    <div
      class="cxd-Page-content"
    >
      <div
        class="cxd-Page-main"
      >
        <div
          class="cxd-Page-body"
          role="page-body"
        >
          <button
            class="cxd-Button cxd-Button--default cxd-Button--size-default"
            type="button"
          >
            <span>
              打开弹窗
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="amis-dialog-widget cxd-Modal cxd-Modal--1th"
    role="dialog"
  >
    <div
      class="cxd-Modal-overlay in"
    />
    <div
      class="cxd-Modal-content in"
    >
      <div
        class="cxd-Modal-header"
      >
        <a
          class="cxd-Modal-close"
          data-position="left"
          data-tooltip="关闭"
        >
          <icon-mock
            classname="icon icon-close"
            icon="close"
          />
        </a>
        <div
          class="cxd-Modal-title"
        >
          模态弹窗
        </div>
      </div>
      <div
        class="cxd-Modal-body"
        role="dialog-body"
      >
        <button
          class="cxd-Button cxd-Button--default cxd-Button--size-default"
          type="button"
        >
          <span>
            打开子弹窗
          </span>
        </button>
        <button
          class="cxd-Button cxd-Button--default cxd-Button--size-default ml-2"
          type="button"
        >
          <span>
            关闭当前弹窗
          </span>
        </button>
        <button
          class="cxd-Button cxd-Button--default cxd-Button--size-default ml-2"
          type="button"
        >
          <span>
            触发确认
          </span>
        </button>
        <button
          class="cxd-Button cxd-Button--default cxd-Button--size-default ml-2"
          type="button"
        >
          <span>
            触发取消
          </span>
        </button>
      </div>
      <div
        class="cxd-Modal-footer"
      >
        <button
          class="cxd-Button cxd-Button--default cxd-Button--size-default"
          type="button"
        >
          <span>
            取消
          </span>
        </button>
        <button
          class="cxd-Button cxd-Button--primary cxd-Button--size-default"
          type="button"
        >
          <span>
            确认
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`2. EventAction:dialog 2`] = `
<div>
  <div
    class="cxd-Page"
  >
    <div
      class="cxd-Page-content"
    >
      <div
        class="cxd-Page-main"
      >
        <div
          class="cxd-Page-body"
          role="page-body"
        >
          <button
            class="cxd-Button cxd-Button--default cxd-Button--size-default"
            type="button"
          >
            <span>
              打开弹窗
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`2. EventAction:dialog 3`] = `
<div>
  <div
    class="cxd-Page"
  >
    <div
      class="cxd-Page-content"
    >
      <div
        class="cxd-Page-main"
      >
        <div
          class="cxd-Page-body"
          role="page-body"
        >
          <button
            class="cxd-Button cxd-Button--default cxd-Button--size-default"
            type="button"
          >
            <span>
              打开弹窗
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`2. EventAction:dialog 4`] = `
<div>
  <div
    class="cxd-Page"
  >
    <div
      class="cxd-Page-content"
    >
      <div
        class="cxd-Page-main"
      >
        <div
          class="cxd-Page-body"
          role="page-body"
        >
          <button
            class="cxd-Button cxd-Button--default cxd-Button--size-default"
            type="button"
          >
            <span>
              打开弹窗
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`2. EventAction:dialog 5`] = `
<div>
  <div
    class="cxd-Page"
  >
    <div
      class="cxd-Page-content"
    >
      <div
        class="cxd-Page-main"
      >
        <div
          class="cxd-Page-body"
          role="page-body"
        >
          <button
            class="cxd-Button cxd-Button--default cxd-Button--size-default"
            type="button"
          >
            <span>
              打开弹窗
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`2. EventAction:dialog 6`] = `
<div>
  <div
    class="cxd-Page"
  >
    <div
      class="cxd-Page-content"
    >
      <div
        class="cxd-Page-main"
      >
        <div
          class="cxd-Page-body"
          role="page-body"
        >
          <button
            class="cxd-Button cxd-Button--default cxd-Button--size-default"
            type="button"
          >
            <span>
              打开弹窗
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`2. EventAction:dialog 7`] = `
<div>
  <div
    class="cxd-Page"
  >
    <div
      class="cxd-Page-content"
    >
      <div
        class="cxd-Page-main"
      >
        <div
          class="cxd-Page-body"
          role="page-body"
        >
          <button
            class="cxd-Button cxd-Button--default cxd-Button--size-default"
            type="button"
          >
            <span>
              打开弹窗
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
